src/components/Header/ConnectMenu/ConnectButton.tsx

```diff
@@ -0,0 +1,52 @@
+import { useRef, useState } from 'react';
+import { useWeb3Context } from '../../../hooks';
+
+import CustomButton from 'src/components/Button/CustomButton';
+import PendingPopper from './PendingPopper';
+import { ReactComponent as MetamaskIcon } from '../../../assets/icons/metamask.svg';
+import { useMediaQuery } from '@material-ui/core';
+
+import { mobileMediaQuery } from 'src/themes/mediaQuery';
+
+export enum ConnectButtonStatus {
+  Connected = 'connected',
+  InProgress = 'inProgress',
+  NotConnected = 'notConnected',
+}
+
+interface ConnectButtonProps {
+  status: ConnectButtonStatus;
+}
+const ConnectButton = ({ status }: ConnectButtonProps) => {
+  const isMobile = useMediaQuery(mobileMediaQuery);
+  const { connect, disconnect } = useWeb3Context();
+  if (status === ConnectButtonStatus.Connected) {
+    if (isMobile) return <CustomButton type="outline" text="Disconnect" onClick={disconnect} />;
```

Comment:
the text need to apply i18n

Reply:
fixed 5a0c6c1

descs haven't been translated will be handle in other ticket. Here I just add todo comments for future modification.